### ISSUE TITLE
how can get simulation finish event?

### ISSUE THREAD

**Issue by swizardlv** [mapbox/guidance-sim#8]
Is there any good way to reset map to before simulation?

Thanks.

**Comment by troyibm** [comment]
I think you might be asking about how to clear some addition to the map that you have done, but clear them all.  Like say for instance you have some markers added to the map that are "permanent" and then you have markers that are added dynamically.  So you might be asking, is there an easy way to remove a select set of markers from the map.

**Comment by swizardlv** [comment]
@troyibm no troy, i just want to reset pitch,zoom and resetNorth when simulation finished. I can remove markers.

**Comment by emilymdubois** [comment]
@swizardlv, thanks for your question! You can reset the simulator by adding an event emitter in the simulator, and conditional handling for that event in your index file.

#### Adding the event emitter to the simulator

Above [this line](https://github.com/mapbox/guidance-sim/blob/c78dd172ceb9a3a8de45bf1c703b0c14c9d824df/lib/simulate.js#L106), you can add an event emitter that communications that the simulation has ended:

```

**Comment by emilymdubois** [comment]
@swizardlv, thanks for your question! You can reset the simulator by adding an event emitter in the simulator, and conditional handling for that event in your index file.

#### Adding the event emitter to the simulator

Above [this line](https://github.com/mapbox/guidance-sim/blob/c78dd172ceb9a3a8de45bf1c703b0c14c9d824df/lib/simulate.js#L106), you can add an event emitter that communications that the simulation has ended:

```
options.response.emit('end');
```

#### Handling the event in your index file

In your index file, which may resemble the [example simulation](https://github.com/mapbox/guidance-sim/blob/c78dd172ceb9a3a8de45bf1c703b0c14c9d824df/sim/index.js), you can add some handling at the [end of the `style.load` section](https://github.com/mapbox/guidance-sim/blob/c78dd172ceb9a3a8de45bf1c703b0c14c9d824df/sim/index.js#L78):

```
  res.on('end', function () {
    map.easeTo({
      center: <desired_start_coordinates>,
      bearing: <desired_start_bearing>,
      duration: <desired_duration_in_ms>,
      pitch: <desired_pitch>,
      zoom: <desired_zoom>,
      easing: function (v) { return v; }
    });
  });
```

I hope that helps! Let me know if you have any more questions.

**Comment by swizardlv** [comment]
@emilymdubois emily, thanks for replying. I think that would fix my issue.could you please have this in your release of guidance-sim？ I don’t want to lost sync with your code

**Comment by swizardlv** [comment]
@emilymdubois may I know how to clean the route path from the map after the simulation?

**Comment by emilymdubois** [comment]
@swizardlv, I merged your PR and rolled a [guidance-sim@0.1.5 release](https://www.npmjs.com/package/guidance-sim).

As for cleaning up the route from your style, you're going to need to reinitialize your map style with the `route*` layers stripped out of it. You can use the `stylePrep()` function available in the [guidance-geojson](https://github.com/mapbox/guidance-geojson) repository to accomplish this. This function strips out any style layers that begin with the string specified in the second parameter. In the following example, it would remove any layers titled `route*`:

```
res.on('end', function () {
  process.env.pause = true;
  map.easeTo({
    center: [-77.051461,38.867035],
    bearing: 180,
    duration: 1000,
    pitch: config.pitch,
    zoom: config.zoom,
    easing: function (v) { return v; }
  }).setStyle(stylePrep(style, 'route'));
});
```

If your simulation file is set up similarly to the example, the simulator [initializes when the style is loaded](https://github.com/mapbox/guidance-sim/blob/c78dd172ceb9a3a8de45bf1c703b0c14c9d824df/sim/index.js#L47). There are probably many ways to circumvent this, but I added `process.env.pause = true` to my `res.on('end)` function above, and the following at the top of my `map.on('style.load')` function:

```
if (process.env.pause === true) {
  process.exit(1)
};
```

I hope that helps!

**Comment by swizardlv** [comment]
@emilymdubois thank you very much!
